### PR TITLE
more refactories

### PR DESCRIPTION
**Pull request by caarlos0**
thanks @aureliojargas

refs #41

### CODE REVIEW

File: jvm.sh [hunk 1]
```diff
@@ -52,27 +52,38 @@ __jvm_pomversion() {
...
   version="$(__jvm_version)"
-  test ! -z "$version" && __jvm_set "$version"
+  test -n "$version" && __jvm_set "$version"
 }
 
 # edit custom java version configurations
 __jvm_config() {
-  test ! -f ~/.jvmconfig && echo "custom-jdk=/path/to/custom/jdk" > ~/.jvmconfig
-  $EDITOR ~/.jvmconfig
+  file="$HOME/.jvmconfig"
```
Comment (aureliojargas) [inline jvm.sh:84]: The `local` keyword is missing

File: jvm.sh [hunk 2]
```diff
@@ -52,27 +52,38 @@ __jvm_pomversion() {
...
   test ! -f .java-version && __jvm_pomversion
-  test -f .java-version && version="$(cat .java-version)"
-  test -z "$version" -a -f ~/.java-version && version="$(cat ~/.java-version)"
-  echo "$version"
+  __jvm_local_version || __jvm_user_version
 }
 
 # called when a dir changes. Find which java version to use and sets it to PATH.
 __jvm_main() {
   version="$(__jvm_version)"
```
Comment (aureliojargas) [inline jvm.sh:78]: The `local` keyword is missing

File: jvm.sh [hunk 3]
```diff
@@ -52,27 +52,38 @@ __jvm_pomversion() {
   test -f pom.xml || return 1
   version="$(__jvm_pomversion_regex)"
```
Comment (aureliojargas) [inline jvm.sh:53]: The `local` keyword is missing

File: jvm.sh (hunk 1 above)
Comment (caarlos0) [inline jvm.sh:84]: not supported in standard shell :/

Comment (aureliojargas) [inline jvm.sh:84]: Oh, you're right. I always forget about it :/

**Pull request merged**